bin/node-template/node/src/service.rs

```diff
@@ -203,13 +201,13 @@ pub fn new_full(config: Configuration)
 		)?;
 	}
 
-	Ok(KeepAliveServiceComponents {
-		task_manager, other: Box::new((telemetry, base_path, rpc)),
-	})
+	task_manager.keep_alive((telemetry, base_path, rpc));
```

Comment:
**I might be wrong...** but I believe it should be possible to move this to sc-service directly.

My reasoning is that this particular piece of code is only relevant for substrate itself. It won't change per project based on substrate. And since it's sc-service who *builds* the `task_manager` based on `Configuration` (provided by value I think), you should be able to get all the things from there and not have a `fn keep_alive()` in the API at all (nor anything related). So it can truly be an implementation detail.